src/main/java/com/ipcounter/classes/CounterStorage.java

```diff
@@ -0,0 +1,58 @@
+package com.ipcounter.classes;
+
+import java.util.BitSet;
+
+/**
+ * Class that can store any number of unique valid IPs
+ */
+public final class CounterStorage {
```

Comment:
finalization says that we can't change or improve behavior of this class. Is it really needed?